### ISSUE TITLE
Release 4.6.0 - Alpha 1 - E2E UX tests - Amazon Cloudtrail integration

### ISSUE THREAD

**Issue by davidjiglesias** [wazuh/wazuh#18830]
## End-to-End (E2E) Testing Guideline

- **Documentation:** Always consult the development documentation for the current stage tag at [this link](https://documentation-dev.wazuh.com/v4.6.0-alpha1/index.html). Be careful because some of the description steps might refer to a current version in production, always navigate using the current development documention for the stage under test.
- **Test Requirements:** Ensure your test comprehensively includes a full stack and agent/s deployment as per the Deployment requirements, detailing the machine OS, installed version, and revision.
- **Deployment Options:** While deployments can be local (using VMs, Vagrant, or Docker) or on the aws-dev account, opt for local deployments when feasible. For AWS access, coordinate with the CICD team through [this link](https://github.com/wazuh/internal-devel-requests/issues/new?assignees=&labels=level%2Ftask%2C+request%2Foperational%2C+type%2Fchange&projects=&template=operational--request.md&title=%3CTitle%3E).
- **External Accounts:** If tests require third-party accounts (e.g., GitHub, Azure, AWS, GCP), request the necessary access through the CICD team [here](https://github.com/wazuh/internal-devel-requests/issues/new?assignees=&labels=level%2Ftask%2C+request%2Foperational%2C+type%2Fchange&projects=&template=operational--request.md&title=%3CTitle%3E).
- **Alerts:** Every test should generate a minimum of one end-to-end alert, from the agent to the dashboard, irrespective of test type.
- **Multi-node Testing:** For multi-node wazuh-manager tests, ensure agents are connected to both workers and the master node.
- **Package Verification:** Use the pre-release package that matches the current TAG you're testing. Confirm its version and revision.
- **Filebeat Errors:** If you encounter errors with Filebeat during testing, refer to [this Slack discussion](https://wazuh-team.slack.com/archives/C03BDG0K6JC/p1672168163537809) for insights and resolutions.
- **Known Issues:** Familiarize yourself with previously reported issues in the Known Issues section. This helps in identifying already recognized errors during testing.
- **Reporting New Issues:** Any new errors discovered during testing that aren't listed under Known Issues should be reported. Communicate these to QA via the c-release Slack channel.
- **Test Conduct:** It's imperative to be thorough in your testing, offering enough detail for reviewers. Incomplete tests might necessitate a redo.
- **Documentation Feedback:** Encountering documentation gaps, unclear guidelines, or anything that disrupts the testing or UX? Open an issue, especially if it's not listed under Known Issues.
- **Format:** If this is your first time doing this, refer to the format (but not necessarily the content, as it may vary) of previous E2E tests, here you have an example https://github.com/wazuh/wazuh/issues/13994.
- **Status and completion:** Change the issue status within your team project accordingly. Once you finish testing and write the conclusions, move it to Pending review and notify the @wazuh/framework team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ). Beware that the reviewers might request additional information or task repetitions.
- **For reviewers:** Please move the issue to Pending final review and notify via Slack using the same thread if everything is ok, otherwise, perform an issue update with the requested changes and move it to On hold, increase the review_cycles in the team project by one and notify the issue assignee via Slack using the same thread.

For the conclusions and the issue testing and updates, use the following legend:

**Status legend**
- 🟢 All checks passed
- 🟡 Found a known issue
- 🔴 Found a new error

## Deployment requirements

| Component | Installation | Type | OS |
|----------|--------------|------|----|
| Indexer | [Quickstart](https://documentation-dev.wazuh.com/v4.6.0-alpha1/quickstart.html) | - | Debian 12 x86_64 |
| Server | [Quickstart](https://documentation-dev.wazuh.com/v4.6.0-alpha1/quickstart.html) | - | Debian 12 x86_64 |
| Dashboard | [Quickstart](https://documentation-dev.wazuh.com/v4.6.0-alpha1/quickstart.html) | - | Debian 12 x86_64 |
| Agent | [Installing Wazuh agents](https://documentation-dev.wazuh.com/v4.6.0-alpha1/installation-guide/wazuh-agent/index.html) | - | Debian 12 x86_64 |

## Test description

Configure AWS Cloudtrail in a Wazuh Manager and a Wazuh Agent.
Ensure the events are correctly displayed on the dashboard.
Test that this proof of concept works as expected for the current release under test:
https://documentation-dev.wazuh.com/v4.6.0-alpha1/proof-of-concept-guide/aws-infrastructure-monitoring.html

Check this documentation page for additional use case examples:
https://documentation-dev.wazuh.com/v4.6.0-alpha1/amazon/services/supported-services/cloudtrail.html

Use discard_regex option to effectively discard certain types of events. 

## Known issues

- https://github.com/wazuh/wazuh-kibana-app/issues/4978
- https://github.com/wazuh/wazuh/issues/15807
- https://github.com/wazuh/wazuh-documentation/issues/4990

<html>
<body>
<h2 dir="auto" style="box-sizing: border-box; margin-top: 24px; margin-bottom: 16px; font-size: 1.5em; font-weight: var(--base-text-weight-semibold, 600); line-height: 1.25; padding-bottom: 0.3em; border-bottom: 1px solid var(--borderColor-muted, var(--color-border-muted)); color: rgb(201, 209, 217); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Conclusions</h2><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(201, 209, 217); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"></p>

Status | Test | Failure type | Notes
-- | -- | -- | --
🟢 | Wazuh Indexer Deployment | - | -
🟢 | Wazuh Server Deployment | - | -
🟢 | Wazuh Dashboard Deployment | - | -
🟢 | Wazuh Agent Deployment | - | -
🟢 | Configure AWS Cloudtrail in Wazuh Manager | - | -
🔴 | Configure AWS Cloudtrail in Wazuh Agent | Cloud trail events do not show on the Wazuh dashboard after integration is done through agent. Some package versions specified in the v4.6.0 documentation are not available when installing in Debian environment. | New issue: https://github.com/wazuh/wazuh/issues/18958, New issue: [#6470](https://github.com/wazuh/wazuh-documentation/issues/6470)
🟢 | CloudTrail events display on dashboard | - | -

</body>
</html>

## Feedback

We value your feedback. Please provide insights on your testing experience.

Was the testing guideline clear? Were there any ambiguities?
- The testing guideline was clear. However, the documentations were not all clear.

Did you face any challenges not covered by the guideline?
- None

Suggestions for improvement:
- Test description can be better itemized as a checklist.t:

## Reviewers validation

The criteria for completing this task is based on the validation of the conclusions and the test results by all reviewers.

All the checkboxes below must be marked in order to close this issue.
- [ ] @davidjiglesias
- [x] @wazuh/framework

**Comment by SamsonIdowu** [comment]
### Wazuh Indexer, Server, Dashboard Deployment 🟢
To do this, a Debian v12.1.0 virtual machine was installed locally. An all in one deployment was done on the virtual machine using the Quickstart script.

1. OS version check
```
root@wazuh-4:~# cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
NAME="Debian GNU/Linux"
VERSION_ID="12"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

2. Wazuh Installation Using Quickstart
`# curl -sO https://packages-dev.wazuh.com/4.6/wazuh-install.sh && sudo bash ./wazuh-install.sh -a`

```
root@wazuh-4:~# curl -sO https://packages-dev.wazuh.com/4.6/wazuh-install.sh && sudo bash ./wazuh-install.sh -a
08/09/2023 21:31:09 INFO: Starting Wazuh installation assistant. Wazuh version: 4.6.0
08/09/2023 21:31:09 INFO: Verbose logging redirected to /var/log/wazuh-install.log
08/09/2023 21:31:17 ERROR: The recommended systems are: Red Hat Enterprise Linux 7, 8, 9; CentOS 7, 8; Amazon Linux 2; Ubuntu 16.04, 18.04, 20.04, 22.04. The current system does not match this list. Use -i|--ignore-check to skip this check.
root@wazuh-4:~#
root@wazuh-4:~# curl -sO https://packages-dev.wazuh.com/4.6/wazuh-install.sh && sudo bash ./wazuh-install.sh -a -i
08/09/2023 21:31:28 INFO: Starting Wazuh installation assistant. Wazuh version: 4.6.0
08/09/2023 21:31:28 INFO: Verbose logging redirected to /var/log/wazuh-install.log
08/09/2023 21:31:47 WARNING: Hardware and system checks ignored.
08/09/2023 21:31:47 INFO: Wazuh web interface port will be 443.
08/09/2023 21:32:00 INFO: Wazuh development repository added.
08/09/2023 21:32:00 INFO: --- Configuration files ---
08/09/2023 21:32:00 INFO: Generating configuration files.
08/09/2023 21:32:06 INFO: Created wazuh-install-files.tar. It contains the Wazuh cluster key, certificates, and passwords necessary for installation.
08/09/2023 21:32:06 INFO: --- Wazuh indexer ---
08/09/2023 21:32:06 INFO: Starting Wazuh indexer installation.
08/09/2023 21:33:43 INFO: Wazuh indexer installation finished.
08/09/2023 21:33:43 INFO: Wazuh indexer post-install configuration finished.
08/09/2023 21:33:43 INFO: Starting service wazuh-indexer.
08/09/2023 21:34:15 INFO: wazuh-indexer service started.
08/09/2023 21:34:15 INFO: Initializing Wazuh indexer cluster security settings.
08/09/2023 21:34:27 INFO: Wazuh indexer cluster initialized.
08/09/2023 21:34:27 INFO: --- Wazuh server ---
08/09/2023 21:34:27 INFO: Starting the Wazuh manager installation.
08/09/2023 21:39:26 INFO: Wazuh manager installation finished.
08/09/2023 21:39:26 INFO: Starting service wazuh-manager.
08/09/2023 21:39:55 INFO: wazuh-manager service started.
08/09/2023 21:39:55 INFO: Starting Filebeat installation.
08/09/2023 21:40:03 INFO: Filebeat installation finished.
08/09/2023 21:40:07 INFO: Filebeat post-install configuration finished.
08/09/2023 21:40:07 INFO: Starting service filebeat.
08/09/2023 21:40:11 INFO: filebeat service started.
08/09/2023 21:40:11 INFO: --- Wazuh dashboard ---
08/09/2023 21:40:14 INFO: --- Dependencies ----
08/09/2023 21:40:14 INFO: Installing chromium-browser.
08/09/2023 21:40:14 WARNING: Cannot install optional dependency: chromium-browser.
08/09/2023 21:40:14 WARNING: Wazuh dashboard dependencies skipped. PDF report generation may not work.
08/09/2023 21:40:14 INFO: Starting Wazuh dashboard installation.
08/09/2023 21:41:44 INFO: Wazuh dashboard installation finished.
08/09/2023 21:41:44 INFO: Wazuh dashboard post-install configuration finished.
08/09/2023 21:41:44 INFO: Starting service wazuh-dashboard.
08/09/2023 21:41:46 INFO: wazuh-dashboard service started.
08/09/2023 21:42:26 INFO: Initializing Wazuh dashboard web application.
08/09/2023 21:42:27 INFO: Wazuh dashboard web application not yet initialized. Waiting...
08/09/2023 21:42:44 INFO: Wazuh dashboard web application not yet initialized. Waiting...
08/09/2023 21:42:59 INFO: Wazuh dashboard web application initialized.
08/09/2023 21:42:59 INFO: --- Summary ---
08/09/2023 21:42:59 INFO: You can access the web interface https://<wazuh-dashboard-ip>:443
    User: admin
    Password: 7qyrsPxx*IVL.*kfqjp5c5n.eHro.T*8
08/09/2023 21:42:59 INFO: Installation finished.
root@wazuh-4:~#
```

### Agent Deployment 🔴 
To do this, a Debian v12.01 virtual machine was installed locally.
Agent version 4.5.2 was installed despite using the v4.6.0 [documentation](https://documentation-dev.wazuh.com/v4.6.0-alpha1/installation-guide/wazuh-agent/wazuh-agent-package-linux.html)

1. OS version check 
```
root@Debian:~# cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
NAME="Debian GNU/Linux"
VERSION_ID="12"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

2. Deploying Wazuh agent
```
root@Debian:~# curl -s https://packages.wazuh.com/key/GPG-KEY-WAZUH | gpg --no-default-keyring --keyring gnupg-ring:/usr/share/keyrings/wazuh.gpg --import && chmod 644 /usr/share/keyrings/wazuh.gpg
gpg: key 96B3EE5F29111145: "Wazuh.com (Wazuh Signing Key) <support@wazuh.com>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
root@Debian:~#
root@Debian:~# echo "deb [signed-by=/usr/share/keyrings/wazuh.gpg] https://packages.wazuh.com/4.x/apt/ stable main" | tee -a /etc/apt/sources.list.d/wazuh.list
deb [signed-by=/usr/share/keyrings/wazuh.gpg] https://packages.wazuh.com/4.x/apt/ stable main
root@Debian:~#
root@Debian:~# apt-get update
Hit:1 http://deb.debian.org/debian bookworm InRelease
Hit:2 http://security.debian.org/debian-security bookworm-security InRelease
Hit:3 http://deb.debian.org/debian bookworm-updates InRelease
Hit:4 https://packages.wazuh.com/4.x/apt stable InRelease
Reading package lists... Done
root@Debian:~#
root@Debian:~# WAZUH_MANAGER="192.168.186.20" apt-get install wazuh-agent
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following NEW packages will be installed:
  wazuh-agent
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 8,943 kB of archives.
After this operation, 30.0 MB of additional disk space will be used.
Get:1 https://packages.wazuh.com/4.x/apt stable/main amd64 wazuh-agent amd64 4.5.2-1 [8,943 kB]
Fetched 8,943 kB in 19s (481 kB/s)
Preconfiguring packages ...
Selecting previously unselected package wazuh-agent.
(Reading database ... 155501 files and directories currently installed.)
Preparing to unpack .../wazuh-agent_4.5.2-1_amd64.deb ...
Unpacking wazuh-agent (4.5.2-1) ...
Setting up wazuh-agent (4.5.2-1) ...
root@Debian:~#
root@Debian:~# systemctl daemon-reload
systemctl enable wazuh-agent
systemctl start wazuh-agent
Synchronizing state of wazuh-agent.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install enable wazuh-agent
root@Debian:~#
root@Debian:~# systemctl status wazuh-agent
● wazuh-agent.service - Wazuh agent
     Loaded: loaded (/lib/systemd/system/wazuh-agent.service; enabled; preset: enabled)
     Active: active (running) since Fri 2023-09-08 21:10:19 MSK; 9s ago
    Process: 14238 ExecStart=/usr/bin/env /var/ossec/bin/wazuh-control start (code=exited, status=0/SUCCESS)
      Tasks: 27 (limit: 4582)
     Memory: 14.8M
        CPU: 1.367s
     CGroup: /system.slice/wazuh-agent.service
             ├─14264 /var/ossec/bin/wazuh-execd
             ├─14275 /var/ossec/bin/wazuh-agentd
             ├─14288 /var/ossec/bin/wazuh-syscheckd
             ├─14301 /var/ossec/bin/wazuh-logcollector
             └─14317 /var/ossec/bin/wazuh-modulesd

Sep 08 21:10:11 Debian systemd[1]: Starting wazuh-agent.service - Wazuh agent...
Sep 08 21:10:11 Debian env[14238]: Starting Wazuh v4.5.2...
Sep 08 21:10:12 Debian env[14238]: Started wazuh-execd...
Sep 08 21:10:14 Debian env[14238]: Started wazuh-agentd...
Sep 08 21:10:15 Debian env[14238]: Started wazuh-syscheckd...
Sep 08 21:10:16 Debian env[14238]: Started wazuh-logcollector...
Sep 08 21:10:17 Debian env[14238]: Started wazuh-modulesd...
Sep 08 21:10:19 Debian env[14238]: Completed.
Sep 08 21:10:19 Debian systemd[1]: Started wazuh-agent.service - Wazuh agent.
```

### Configure AWS Cloudtrail in a Wazuh Manager and a Wazuh Agent. 🟢

To do this, the [Configure AWS Cloudtrail v4.6.0 document](https://documentation-dev.wazuh.com/v4.6.0-alpha1/proof-of-concept-guide/aws-infrastructure-monitoring.html) was used.

1. CloudTrail and S3 bucket was created by CICD team and details were shared.

2. The **Amazon AWS module** was enabled on the Wazuh v4.6.0 dashboard by navigating to **Settings** > **Modules**. 
<img width="953" alt="image" src="https://github.com/wazuh/wazuh/assets/66468924/f0271c53-27db-42f1-b15a-582524e89933">

3. To provide AWS authentication details, the following configuration was added to the `/root/.aws/credentials` file on the Wazuh v4.6.0 server. [Ref](https://documentation-dev.wazuh.com/v4.6.0-alpha1/cloud-security/amazon/services/prerequisites/credentials.html#aws-profile):
```
[default]
aws_access_key_id=AKIAYIPNU4F*************
aws_secret_access_key=7/oI9eY3+6VVNX***********************
region=us-east-1

[dev]
aws_access_key_id=foo2
aws_secret_access_key=bar2
region=us-east-1

[prod]
aws_access_key_id=foo2
aws_secret_access_key=bar2
region=us-east-1
```

4. The following configuration was added to `/var/ossec/etc/ossec.conf` file on the Wazuh v4.6.0 server. [Ref](https://documentation-dev.wazuh.com/v4.6.0-alpha1/proof-of-concept-guide/aws-infrastructure-monitoring.html):
```
  <wodle name="aws-s3">
    <disabled>no</disabled>
    <interval>30m</interval>
    <run_on_start>yes</run_on_start>
    <skip_on_error>no</skip_on_error>

    <bucket type="cloudtrail">
      <name>aws-cloudtrail-logs-567970947422-a350a676</name>
      <aws_profile>default</aws_profile>
    </bucket>
  </wodle>
```

5. Wazuh server was restarted
```
root@wazuh-dashboard-test:~# sudo systemctl restart wazuh-manager
root@wazuh-dashboard-test:~# systemctl status wazuh-manager
● wazuh-manager.service - Wazuh manager
     Loaded: loaded (/lib/systemd/system/wazuh-manager.service; enabled; preset: enabled)
     Active: active (running) since Fri 2023-09-08 22:03:39 MSK; 10s ago
    Process: 71277 ExecStart=/usr/bin/env /var/ossec/bin/wazuh-control start (code=exited, status=0/SUCCESS)
      Tasks: 141 (limit: 4582)
     Memory: 382.6M
        CPU: 41.183s
     CGroup: /system.slice/wazuh-manager.service
             ├─71335 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh-apid.py
             ├─71336 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh-apid.py
             ├─71339 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh-apid.py
             ├─71342 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh-apid.py
             ├─71383 /var/ossec/bin/wazuh-authd
             ├─71400 /var/ossec/bin/wazuh-db
             ├─71424 /var/ossec/bin/wazuh-execd
             ├─71438 /var/ossec/bin/wazuh-analysisd
             ├─71449 /var/ossec/bin/wazuh-syscheckd
             ├─71516 /var/ossec/bin/wazuh-remoted
             ├─71548 /var/ossec/bin/wazuh-logcollector
             ├─71567 /var/ossec/bin/wazuh-monitord
             ├─71591 /var/ossec/bin/wazuh-modulesd
             ├─71606 /bin/sh wodles/aws/aws-s3 --bucket aws-cloudtrail-logs-567970947422-a350a676 --aws_profile default --type cloudtrail
             └─71618 /var/ossec/framework/python/bin/python3 /var/ossec/wodles/aws/aws-s3.py --bucket aws-cloudtrail-logs-567970947422-a350a676 --aws_profile default --type cloud>

Sep 08 22:03:29 wazuh-dashboard-test env[71277]: Started wazuh-db...
Sep 08 22:03:30 wazuh-dashboard-test env[71277]: Started wazuh-execd...
Sep 08 22:03:31 wazuh-dashboard-test env[71277]: Started wazuh-analysisd...
Sep 08 22:03:32 wazuh-dashboard-test env[71277]: Started wazuh-syscheckd...
Sep 08 22:03:33 wazuh-dashboard-test env[71277]: Started wazuh-remoted...
Sep 08 22:03:34 wazuh-dashboard-test env[71277]: Started wazuh-logcollector...
Sep 08 22:03:35 wazuh-dashboard-test env[71277]: Started wazuh-monitord...
Sep 08 22:03:36 wazuh-dashboard-test env[71277]: Started wazuh-modulesd...
Sep 08 22:03:39 wazuh-dashboard-test env[71277]: Completed.
Sep 08 22:03:39 wazuh-dashboard-test systemd[1]: Started wazuh-manager.service - Wazuh manager.
```

6. CloudTrail events could be seen in the Amazon AWS module on the Wazuh dashboard.
<img width="956" alt="image" src="https://github.com/wazuh/wazuh/assets/66468924/524cb1d9-ef61-47c0-adc4-52143525c6e1">

<img width="954" alt="image" src="https://github.com/wazuh/wazuh/assets/66468924/01e1157f-7be6-4b31-a15b-752cf5a2f452">

7. An observation was made regarding high latency in delivery of CloudTrail events to the Wazuh Server. (*This could have been caused by a number of factors ranging from network, to compute resources available to the Wazuh server.*)



### Conclusions

All tests have been executed and a conclusion of the results can be found here.

🟢	Wazuh Indexer Deployment	
🟢	Wazuh Server Deployment		
🟢	Wazuh Dashboard Deployment	
🔴 	Wazuh Agent Deployment - v4.5.2 package was deployed instead of v4.6.0
🟢	AWS CloudTrail POC
🟢	AWS CloudTrail events display on dashboard


### Summary
The CloudTrail POC was implemented successfully.

**Comment by nico-stefani** [comment]
@SamsonIdowu please open a new documentation issue to specify a different version of `numpy`.
Regarding the events from the agent, attach logs to show the malfunction and open an issue if necessary.

**Comment by SamsonIdowu** [comment]
> @SamsonIdowu please open a new documentation issue to specify a different version of `numpy`. Regarding the events from the agent, attach logs to show the malfunction and open an issue if necessary.

An issue has been raised regarding the `numphy` version with documentation team https://github.com/wazuh/wazuh/issues/18948.

**Comment by nico-stefani** [comment]
> > @SamsonIdowu please open a new documentation issue to specify a different version of `numpy`. Regarding the events from the agent, attach logs to show the malfunction and open an issue if necessary.
> 
> An issue has been raised regarding the `numphy` version with documentation team #18948.

The issue needs to be in https://github.com/wazuh/wazuh-documentation.
Also, could you check the logs from the agent?

**Comment by SamsonIdowu** [comment]
> > > @SamsonIdowu please open a new documentation issue to specify a different version of `numpy`. Regarding the events from the agent, attach logs to show the malfunction and open an issue if necessary.
> > 
> > 
> > An issue has been raised regarding the `numphy` version with documentation team #18948.
> 
> The issue needs to be in https://github.com/wazuh/wazuh-documentation. Also, could you check the logs from the agent?

Please see logs from the agent:

```
root@Debian:~# tail -f /var/ossec/logs/ossec.log
2023/09/11 17:34:01 wazuh-modulesd:control: INFO: Starting control thread.
2023/09/11 17:34:01 wazuh-modulesd:syscollector: INFO: Module started.
2023/09/11 17:34:01 wazuh-modulesd:syscollector: INFO: Starting evaluation.
2023/09/11 17:34:01 wazuh-modulesd:aws-s3: WARNING: Bucket:  -  Returned exit code 4
2023/09/11 17:34:01 wazuh-modulesd:aws-s3: WARNING: Bucket:  -  boto3 module is required.

2023/09/11 17:34:01 wazuh-modulesd:aws-s3: INFO: Fetching logs finished.
2023/09/11 17:34:01 wazuh-modulesd:syscollector: INFO: Evaluation finished.
2023/09/11 17:34:03 wazuh-syscheckd: INFO: (6009): File integrity monitoring scan ended.
2023/09/11 17:34:03 wazuh-syscheckd: INFO: FIM sync module started.
2023/09/11 17:35:13 rootcheck: INFO: Ending rootcheck scan.
```

While the log output indicates that the `boto3` module is required, I made sure it is installed.

```
root@Debian:~# pip3 show boto3
Name: boto3
Version: 1.17.85
Summary: The AWS SDK for Python
Home-page: https://github.com/boto/boto3
Author: Amazon Web Services
Author-email:
License: Apache License 2.0
Location: /root/myenv/lib/python3.11/site-packages
Requires: botocore, jmespath, s3transfer
Required-by:
root@Debian:~#
```

**Comment by SamsonIdowu** [comment]
New issue has been raised in documentation https://github.com/wazuh/wazuh-documentation/issues/6470

**Comment by nico-stefani** [comment]
> > > > @SamsonIdowu please open a new documentation issue to specify a different version of `numpy`. Regarding the events from the agent, attach logs to show the malfunction and open an issue if necessary.
> > > 
> > > 
> > > An issue has been raised regarding the `numphy` version with documentation team #18948.
> > 
> > 
> > The issue needs to be in https://github.com/wazuh/wazuh-documentation. Also, could you check the logs from the agent?
> 
> Please see logs from the agent:
> 
> ```
> root@Debian:~# tail -f /var/ossec/logs/ossec.log
> 2023/09/11 17:34:01 wazuh-modulesd:control: INFO: Starting control thread.
> 2023/09/11 17:34:01 wazuh-modulesd:syscollector: INFO: Module started.
> 2023/09/11 17:34:01 wazuh-modulesd:syscollector: INFO: Starting evaluation.
> 2023/09/11 17:34:01 wazuh-modulesd:aws-s3: WARNING: Bucket:  -  Returned exit code 4
> 2023/09/11 17:34:01 wazuh-modulesd:aws-s3: WARNING: Bucket:  -  boto3 module is required.
> 
> 2023/09/11 17:34:01 wazuh-modulesd:aws-s3: INFO: Fetching logs finished.
> 2023/09/11 17:34:01 wazuh-modulesd:syscollector: INFO: Evaluation finished.
> 2023/09/11 17:34:03 wazuh-syscheckd: INFO: (6009): File integrity monitoring scan ended.
> 2023/09/11 17:34:03 wazuh-syscheckd: INFO: FIM sync module started.
> 2023/09/11 17:35:13 rootcheck: INFO: Ending rootcheck scan.
> ```
> 
> While the log output indicates that the `boto3` module is required, I made sure it is installed.
> 
> ```
> root@Debian:~# pip3 show boto3
> Name: boto3
> Version: 1.17.85
> Summary: The AWS SDK for Python
> Home-page: https://github.com/boto/boto3
> Author: Amazon Web Services
> Author-email:
> License: Apache License 2.0
> Location: /root/myenv/lib/python3.11/site-packages
> Requires: botocore, jmespath, s3transfer
> Required-by:
> root@Debian:~#
> ```

Ok, open an issue, in `wazuh/wazuh` with all this information, to investigate and address this possible problem.

**Comment by SamsonIdowu** [comment]
> investigate and address this possible problem

New issue has been opened in `wazuh/wazuh` to investigate and address the problem with  `wazuh-modulesd:aws-s3`:

https://github.com/wazuh/wazuh/issues/18958

**Comment by nico-stefani** [comment]
LGTM!

**Comment by davidjiglesias** [comment]
Please, update feedback and conclusions in the main body of the issue. It is important for the QA team to easily collect and validate all the results and to collect feedback from multiple issues.

**Comment by davidjiglesias** [comment]
LGTM!
